mesonbuild/dependencies/misc.py

```diff
@@ -359,15 +364,19 @@ def detect_lib_modules_win(self):
                     fname = os.path.basename(entry)
                     self.lib_modules[self.modname_from_filename(fname)] = [fname]
 
+    def abi_tag(self):
+        if mesonlib.for_windows(self.want_cross, self.env):
+            return None
+        if self.is_multithreading and mesonlib.for_darwin(self.want_cross, self.env):
+            return '-mt'
+        else:
+            return ''
+
     def detect_lib_modules_nix(self):
         all_found = True
         for module in self.requested_modules:
-            args = None
-            libname = 'boost_' + module
-            if self.is_multithreading and mesonlib.for_darwin(self.want_cross, self.env):
-                # - Linux leaves off -mt but libraries are multithreading-aware.
-                # - Mac requires -mt for multithreading, so should not fall back to non-mt libraries.
-                libname = libname + '-mt'
```

Comment:
These comments are important. Please don't delete them but move them to the abi_tag function instead.

Reply:
Sure, good point.  I put these back added a bit to the documentation earlier in the file.